### ISSUE TITLE
Drag & Drop Task to another Task List

### ISSUE THREAD

**Issue by marbetschar** [elementary/tasks#168]
It should be possible to move a task to another list by using Drag & Drop.

**Comment by alainm23** [comment]
Is there a property in `ECal.Component` or `ICal.Component` to store the order of the task in the list?

**Comment by marbetschar** [comment]
@alainm23 I don't know if there is a standard way of sorting. I can't find anything sort-related in the [RFC](https://tools.ietf.org/html/rfc5545#section-3.6.2). And iOS/Apple uses a custom property called `X-APPLE-SORT-ORDER`.

You can access [custom properties similar to this](https://github.com/elementary/tasks/blob/f3817889b4c78e7cfc9f3a380668ef5f2ccb2c05/src/Util.vala#L156-L195).

**Comment by alainm23** [comment]
@marbetschar For this to work we have to change how the task editing works, currently the [grab_focus](https://github.com/elementary/tasks/blob/master/src/Widgets/TaskRow.vala#L248) event is used to enable the editing of a task. This does not allow a task to drag naturally. In Planner I use the [row_activated](https://github.com/alainm23/planner/blob/master/src/Views/Inbox.vala#L331) event to enable editing of a task, we could make that change first and D&D could work.

**Comment by marbetschar** [comment]
@alainm23 sounds good to me. The current task edit abort handling seems to interfere with row activation anyways.

BTW: I've added some utility functions to access X-Properties in #174 (e.g.: Util.get_icalcomponent_x_property) - so once this is merged it should also be easier to access `X-APPLE-SORT-ORDER` or similar props.
